src/app/page.tsx

```diff
@@ -1,3 +1,5 @@
+'use client';
```

Comment:
Haha so this was a question I was gonna ask you. but reading up on the Next.js [rendering doc](https://nextjs.org/docs/app/building-your-application/rendering), Next.js is opt-in to client-rendered components. Therefore, all components by default are rendered on the server and thus can't handle user state (setting state based on scroll position) like a component rendered on the client-side would. 

I'm just specifying `use client` on the parent component so all child components linked to it will be rendered client-side.

Reply:
Ahh okay cool. I actually have needed to use clientside rendering in my Next.js project before, but I just used `next/dynamic` with the `ssr: false` [option](https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#with-no-ssr). But this is definitely cleaner!

Though it might be worth moving this down to just where you need it so that you can take advantage of the [other benefits of server-side rendering](https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#with-no-ssr)